src/DDTrace/Integrations/Yii/YiiSandboxedIntegration.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace DDTrace\Integrations\Yii;
+
+use DDTrace\Integrations\SandboxedIntegration;
+
+class YiiSandboxedIntegration extends SandboxedIntegration
+{
+    const NAME = 'yii';
+
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+        return self::$instance;
+    }
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return self::NOT_AVAILABLE;
+        }
+
+        $integration = self::getInstance();
+
+        // This happens somewhat early in the setup, though there may be a better candidate
+        \dd_trace_method('yii\di\Container', '__construct', function () use ($integration) {
+            $version = \Yii::getVersion();
+            $majorVersion = \substr($version, 0, 2);
```

Comment:
If Yii follows semver, you could use utilities in `\DDTrace\Util\Versions::versionMatches('2', \Yii::getVersion())`

Reply:
They do not follow semver, but I suspect this will work on all v2 releases anyway.  Think I should switch?